core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -3796,7 +3796,14 @@ class KafkaApis(val requestChannel: RequestChannel,
     if (!isConsumerGroupProtocolEnabled()) {
       // The API is not supported by the "old" group coordinator (the default). If the
       // new one is not enabled, we fail directly here.
-      requestHelper.sendMaybeThrottle(request, consumerGroupHeartbeatRequest.getErrorResponse(Errors.UNSUPPORTED_VERSION.exception))
+      requestHelper.sendMaybeThrottle(request, new ConsumerGroupHeartbeatResponse(
+        new ConsumerGroupHeartbeatResponseData()
+          .setErrorCode(Errors.UNSUPPORTED_VERSION.code())
+          .setErrorMessage("Failed to handle consumer group heartbeat request since the " +
+            "consumer group coordinator is not enabled. Please enable the consumer group coordinator in the brokers configuration by setting " +
+            s"\"${GroupCoordinatorConfig.NEW_GROUP_COORDINATOR_ENABLE_CONFIG}\" to \"true\" and " +
+            s"\"${GroupCoordinatorConfig.GROUP_COORDINATOR_REBALANCE_PROTOCOLS_CONFIG}\" to \"classic, consumer\". " +
+            s"Or setting \"group.protocol\" to \"classic\" in the consumer configuration.")))
```

Comment:
The message is a bit misleading as is because the group coordinator _is_ enabled, it's just that the `group.protocol` Some minor tweaks:

```suggestion
      requestHelper.sendMaybeThrottle(request, new ConsumerGroupHeartbeatResponse(
        new ConsumerGroupHeartbeatResponseData()
          .setErrorCode(Errors.UNSUPPORTED_VERSION.code())
          .setErrorMessage("An error occurred handling the consumer group heartbeat request. " +
            s"To enable support for the ${Group.GroupType.CONSUMER} group protocol, update the broker " +
            s"configuration for \"${GroupCoordinatorConfig.NEW_GROUP_COORDINATOR_ENABLE_CONFIG}\" to \"true\" and " +
            s"\"${GroupCoordinatorConfig.GROUP_COORDINATOR_REBALANCE_PROTOCOLS_CONFIG}\" to " +
            s"\"${Group.GroupType.CLASSIC}, ${Group.GroupType.CONSUMER}\". Alternatively, update the client " +
            s"configuration for \"group.protocol\" to \"${Group.GroupType.CLASSIC}\".")))
```

Reply:
Thanks for the suggestion. Update it.